src/pages/api/chapters/index.ts

```diff
@@ -1,13 +1,29 @@
-import type { NextApiRequest, NextApiResponse } from 'next';
+import type { NextApiResponse } from 'next';
 import { Chapter, PrismaClient, Prisma } from '@prisma/client';
-import { ErrorResponse } from '@/utils/types';
+import { ErrorResponse, serverErrorHandler } from '@/utils/error';
+import { NextIronRequest, withSession } from '@/utils/session';
+import { SessionAdminUser } from '../admin/login';
+import { getPasswordHash } from '@/utils/password';
+import {
+  validateChapterInput,
+  validateNewChapterUserInput,
+  validateNewUserInput,
+} from '@/utils/prisma-validation';
 
 type DataResponse = {
   chapters: Chapter[];
 };
 
-export default async function handler(
-  req: NextApiRequest,
+export type ChapterInputBody = {
+  chapter: Prisma.ChapterCreateInput;
+  newUser?: {
+    chapterUser: Prisma.ChapterUserCreateInput;
```

Comment:
Could you refactor this so that the type of the newUser field is the intersection of the `ChapterUserCreateInput` and `UserCreateInput` types? I think removing the nested object would make things less confusing for the front-end devs